src/Spanner/Session/CacheSessionPool.php

```diff
@@ -777,4 +764,63 @@ private function validateConfig()
             );
         }
     }
+
+    /**
+     * Delete the provided sessions.
+     *
+     * @param array $sessions
+     */
+    private function deleteSessions(array $sessions)
+    {
+        $calls = [];
+
+        foreach ($sessions as $session) {
+            $calls[] = $this->database->connection()
+                ->deleteSessionAsync([
+                    'name' => $session['name'],
+                    'database' => $this->database->name()
+                ]);
+        }
+
+        foreach ($calls as $call) {
+            $call->wait();
```

Comment:
I just ran a small test comparing `DeleteSession` with `GetSession`, using this code snippet:
```php
$call = $stub->DeleteSession($deleteRequest, [], ['call_credentials_callback' => $grpcHelper->createCallCredentialsCallback()]);

$i = 0;
while (true) {
    $fetchedSession = $spanner->getSession($sessionName);
    echo "fetched: $i\n";
    $i += 1;
}
```

I found that this snippet would make approximately 10 successful calls to `getSession` before eventually failing with a `Session not found` error. (This is true even when `DeleteSession` is not the first call on the connection).